code/serializers/Basic/RESTfulAPI_BasicSerializer.php

```diff
@@ -116,82 +112,94 @@ public function serialize($data)
+     *                                is permission denied
      */
     protected function formatDataObject(DataObject $dataObject)
     {
         // api access control
-    if (!RESTfulAPI::api_access_control($dataObject, 'GET')) {
-        return null;
-    }
+        if (!RESTfulAPI::api_access_control($dataObject, 'GET')) {
+            return null;
+        }
 
         if (method_exists($dataObject, 'onBeforeSerialize')) {
             $dataObject->onBeforeSerialize();
         }
         $dataObject->extend('onBeforeSerialize');
 
-    // setup
-    $formattedDataObjectMap = array();
+        // setup
+        $formattedDataObjectMap = array();
 
-    // get DataObject config
-    $db                    = Config::inst()->get($dataObject->ClassName, 'db');
+        // get DataObject config
+        $db                    = Config::inst()->get($dataObject->ClassName, 'db');
         $has_one               = Config::inst()->get($dataObject->ClassName, 'has_one');
         $has_many              = Config::inst()->get($dataObject->ClassName, 'has_many');
         $many_many             = Config::inst()->get($dataObject->ClassName, 'many_many');
         $belongs_many_many     = Config::inst()->get($dataObject->ClassName, 'belongs_many_many');
-    
-    //$many_many_extraFields = $dataObject->many_many_extraFields();
-    $many_many_extraFields = $dataObject->stat('many_many_extraFields');
 
-    // setup ID (not included in $db!!)
-    $serializedColumnName = $this->serializeColumnName('ID');
+        //$many_many_extraFields = $dataObject->many_many_extraFields();
+        $many_many_extraFields = $dataObject->stat('many_many_extraFields');
+
+        // setup ID (not included in $db!!)
+        $serializedColumnName = $this->serializeColumnName('ID');
         $formattedDataObjectMap[$serializedColumnName] = $dataObject->getField('ID');
 
-    // iterate over simple DB fields
-    if (!$db) {
-        $db = array();
-    }
-    $apiFields = (array) Config::inst()->get($dataObject->ClassName, 'api_fields');
-    foreach ($db as $columnName => $fieldClassName) {
-        if (!empty($apiFields) && !in_array($columnName, $apiFields)) {
-            continue;
+        // iterate over simple DB fields
+        if (!$db) {
+            $db = array();
         }
-        $serializedColumnName = $this->serializeColumnName($columnName);
-        $formattedDataObjectMap[$serializedColumnName] = $dataObject->getField($columnName);
-    }
 
-    // iterate over has_one relations
-    if (!$has_one) {
-        $has_one = array();
-    }
-        foreach ($has_one as $columnName => $fieldClassName) {
+        // Get a possibly defined list of "api_fields" for this DataObject. If defined, they will be the only fields
+        // for this DataObject that will be returned, including related models.
+        $apiFields = (array) Config::inst()->get($dataObject->ClassName, 'api_fields');
```

Comment:
Could we have the `api_fields` declaration below the other `db`, `has_one`, `has_many` etc on line 140? Just easier to follow the code and what `api_fields` relates to....